.github/workflows/main.yml

```diff
@@ -50,3 +50,27 @@ jobs:
           provider: microk8s
       - name: Run integration tests
         run: tox -e integration
+      - name: Archive Tested Charm
+        uses: actions/upload-artifact@v3
+        if: ${{ github.event_name == 'push' }}
+        with:
+          name: tested-charm
+          path: .tox/**/tls-certificates-operator_ubuntu-20.04-amd64.charm
+          retention-days: 5
+
+  publish-charm:
+    name: Publish Charm
+    needs: integration-test
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'push' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Fetch Tested Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: tested-charm
+      - name: Upload charm to Charmhub edge channel
```

Comment:
why wouldn't we use `canonical/charming-actions/channel@2.0.0-rc` to select the channel as it was in `publish.yaml`?

Reply:
Good point, I can probably work that back in. The goal is to improve the `charming-actions` afterwards once the mechanics of this workflow are well tested.

Reply:
Fixed